submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/test/submission/submit/async/SubmitAsyncApiTest.kt

```diff
@@ -0,0 +1,93 @@
+import org.assertj.core.api.Assertions.assertThat
+import org.awaitility.Awaitility.await
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.amqp.core.Binding
+import org.springframework.amqp.core.Binding.DestinationType.QUEUE
+import org.springframework.amqp.core.Queue
+import org.springframework.amqp.rabbit.core.RabbitAdmin
+import org.springframework.amqp.rabbit.core.RabbitTemplate
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.web.server.LocalServerPort
+import org.springframework.core.ParameterizedTypeReference
+import org.springframework.web.client.RestTemplate
+import uk.ac.ebi.events.config.BIOSTUDIES_EXCHANGE
+import uk.ac.ebi.events.config.SUBMISSIONS_ROUTING_KEY
+import java.util.concurrent.TimeUnit.SECONDS
+
+private const val QUEUE_NAME = "submissions_queue"
+
+@ExtendWith(TemporaryFolderExtension::class)
+internal class SubmitAsyncApiTest(tempFolder: TemporaryFolder) : BaseIntegrationTest(tempFolder) {
+
+    @Nested
+    @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+    inner class SubmitAsyncTSV(
+        @Autowired private val securityTestService: SecurityTestService,
+        @Autowired private val rabbitTemplate: RabbitTemplate,
+        @Autowired private val rabbitAdmin: RabbitAdmin,
+        @Autowired private val jmsConfig: JmsConfig
+    ) {
+        @LocalServerPort
+        private var serverPort: Int = 0
+
+        private lateinit var webClient: BioWebClient
+        val restTemplate = RestTemplate()
+
+        @BeforeAll
+        fun init() {
+            securityTestService.registerUser(SuperUser)
+            webClient = getWebClient(serverPort, SuperUser)
+            createQueue(QUEUE_NAME, BIOSTUDIES_EXCHANGE, SUBMISSIONS_ROUTING_KEY)
+        }
+
+        @Test
+        fun test() {
+            val submissionTSV = tsv {
+                line("Submission", "S-TEST22")
+                line("Title", "Test title")
+                line("ReleaseDate", "2000-01-31")
+                line()
+            }.toString()
+
+            webClient.submitAsync(submissionTSV, TSV)
+
+            val message = await().atMost(10, SECONDS).untilNotNull {
+                rabbitTemplate.receiveAndConvert(QUEUE_NAME, MapType)
+            }
+
+            val pageTabUrlJSON: String = message.getValue("pagetabUrl").replace("8080", "$serverPort")
+            val pageTabUrlTSV = pageTabUrlJSON.replace("json", "tsv")
+            val result = restTemplate.getForObject(pageTabUrlTSV, String::class.java)
+
+            assertThat(result).isNotNull()
+            assertThat(result.trim()).isEqualTo(submissionTSV.trim())
+        }
+
+        private fun createQueue(queueName: String, exchangeName: String, routingKey: String) {
```

Comment:
I can see ourselves using this method a lot. Perhaps it would be a good idea to add this to the `BaseIntegrationTest` class

Reply:
right now just here , if we start adding more test we should consider it yes. I do consider appropriated to added before is needed or with not clear ensure that we will.